app/helpers/cdx_form_helper.rb

```diff
@@ -61,10 +67,26 @@ def errors_to_show
   end
 
   def error_messages_to_show
-    errors_to_show.map do |attribute|
-      @object.errors.full_messages_for(attribute)
-    end.flatten.tap do
-      @errors_to_show = []
+    Hash[errors_to_show.map do |attribute|
+      messages = @object.errors.full_messages_for(attribute)
+      next if messages.empty?
+      [attribute, messages]
+    end.compact]
+  end
+
+  def log_unhandled_errors
+    unhandled_errors = error_messages_to_show.dup
+    unhandled_errors.delete(:base)
```

Comment:
The following would achieve the same in one action, but I'm not sure it improves readability?

```suggestion
    unhandled_errors = error_messages_to_show.reject { |k, _| k == :base }
```